[CI] DLMFrozenTransitionDisruptionIT testMasterFailoverDuringCleanup failing

**Build Scans:**
- [elasticsearch-intake #43542 / part2](https://gradle-enterprise.elastic.co/s/xbkgj3ayh2vku)
- [elasticsearch-intake #43021 / part2](https://gradle-enterprise.elastic.co/s/znyzcr3yllp52)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:dlm-frozen-transition:internalClusterTest" --tests "org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionDisruptionIT.testMasterFailoverDuringCleanup" -Dtests.seed=CB22EC7A91204BF8 -Dtests.locale=teo -Dtests.timezone=Europe/Moscow -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionDisruptionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testMasterFailoverDuringCleanup'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: disruption thread [test-disruption-indices:admin/data_stream/modify] did not finish in time
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testMasterFailoverDuringCleanup
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testMasterFailoverDuringCleanup (1.3% fail rate in 152 executions)
- [main] 2 failures in step part2 (9.1% fail rate in 22 executions)
- [main] 2 failures in pipeline elasticsearch-intake (9.1% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -640,9 +640,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT
   method: testSourceRelocationAndTargetRestart
   issue: https://github.com/elastic/elasticsearch/issues/153173
-- class: org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionDisruptionIT
-  method: testMasterFailoverDuringCleanup
-  issue: https://github.com/elastic/elasticsearch/issues/153174
 - class: org.elasticsearch.packaging.test.PackageUpgradeTests
   method: test12SetupBwcVersion
   issue: https://github.com/elastic/elasticsearch/issues/152819

--- a/x-pack/plugin/dlm-frozen-transition/src/internalClusterTest/java/org/elasticsearch/xpack/dlm/frozen/DLMFrozenTransitionDisruptionIT.java
+++ b/x-pack/plugin/dlm-frozen-transition/src/internalClusterTest/java/org/elasticsearch/xpack/dlm/frozen/DLMFrozenTransitionDisruptionIT.java
@@ -697,6 +697,17 @@ public class DLMFrozenTransitionDisruptionIT extends ESIntegTestCase {
      * master node while it is being stopped asynchronously by the disruption thread.
      */
     private void waitForStableCluster(int nodeCount) throws Exception {
+        // The disruption latch fires before the async disruption thread finishes stopping the
+        // node. If we probe cluster health while the node is still shutting down,
+        // ensureStableCluster can route the health request via the dying node's local client,
+        // whose response future is never completed (no transport disconnect fires for in-JVM
+        // client calls, and the node's scheduler that would enforce the request timeout is
+        // gone), hanging the test until the suite timeout. Wait for the disruption to actually
+        // finish first.
+        for (Thread t : disruptionThreadsToJoin) {
+            t.join(TimeUnit.SECONDS.toMillis(60));
+            assertFalse("disruption thread [" + t.getName() + "] did not finish in time", t.isAlive());
+        }
         assertBusy(() -> {
             try {
                 ensureStableCluster(nodeCount);

